### PR TITLE
Ensure Skia images are always disposed

### PR DESCRIPTION
**Pull request by JPVenson**
If there are transiant errors while handing skia wrapped images, skia objects are not always disposed correctly.

Fixes:
Possible memory leaks

### CODE REVIEW

File: src/Jellyfin.Drawing.Skia/SplashscreenBuilder.cs [hunk 1]
```diff
@@ -124,24 +134,32 @@ private SKBitmap GenerateCollage(IReadOnlyList<string> posters, IReadOnlyList<st
     private SKBitmap Transform3D(SKBitmap input)
     {
         var bitmap = new SKBitmap(FinalWidth, FinalHeight);
-        using var canvas = new SKCanvas(bitmap);
-        canvas.Clear(SKColors.Black);
-        var matrix = new SKMatrix
+        try
```
Comment (crobibero) [inline src/Jellyfin.Drawing.Skia/SplashscreenBuilder.cs:137]: should follow the pattern here

File: src/Jellyfin.Drawing.Skia/SkiaEncoder.cs [hunk 2]
```diff
@@ -350,43 +379,52 @@ private SKBitmap OrientImage(SKBitmap bitmap, SKEncodedOrigin origin)
...
-                break;
-            case SKEncodedOrigin.LeftBottom:
-                surface.Translate(0, rotated.Height);
-                surface.RotateDegrees(-90);
-                break;
-        }
-
-        surface.DrawBitmap(bitmap, 0, 0);
-        return rotated;
+        try
```
Comment (crobibero) [inline src/Jellyfin.Drawing.Skia/SkiaEncoder.cs:382]: should follow the pattern here

File: src/Jellyfin.Drawing.Skia/SplashscreenBuilder.cs (hunk 1 above)
Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/SplashscreenBuilder.cs:137]: please elaborate.

File: src/Jellyfin.Drawing.Skia/SkiaEncoder.cs (hunk 2 above)
Comment (crobibero) [reply src/Jellyfin.Drawing.Skia/SkiaEncoder.cs:382]:
```c#
        SKBitmap? rotated = null;
        try
        {
            rotated = needsFlip
                ? new SKBitmap(bitmap.Height, bitmap.Width)
                : new SKBitmap(bitmap.Width, bitmap.Height);
```

File: src/Jellyfin.Drawing.Skia/SplashscreenBuilder.cs (hunk 1 above)
Comment (crobibero) [reply src/Jellyfin.Drawing.Skia/SplashscreenBuilder.cs:142]:
```c#
        SKBitmap? bitmap = null;
        try
        {
            bitmap = new SKBitmap(FinalWidth, FinalHeight);
```

**Review by crobibero — APPROVED** [review]

File: src/Jellyfin.Drawing.Skia/SkiaEncoder.cs [hunk 3]
```diff
@@ -269,14 +269,24 @@ private static SKEncodedOrigin GetSKEncodedOrigin(ImageOrientation? orientation)
...
 
-            origin = codec.EncodedOrigin;
+                origin = codec.EncodedOrigin;
 
-            return bitmap;
+                return bitmap!;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Detected intermediary error decoding image {0}", path);
```
Comment (Bond-009) [inline src/Jellyfin.Drawing.Skia/SkiaEncoder.cs:286]: Won't this get logged twice? Once here and once higher up the call stack.

File: src/Jellyfin.Drawing.Skia/SkiaEncoder.cs [hunk 4]
```diff
@@ -286,17 +296,26 @@ private static SKEncodedOrigin GetSKEncodedOrigin(ImageOrientation? orientation)
...
-        }
 
-        origin = SKEncodedOrigin.TopLeft;
-        return resultBitmap;
+            origin = SKEncodedOrigin.TopLeft;
+            return resultBitmap;
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Detected intermediary error decoding image {0}", path);
```
Comment (Bond-009) [inline src/Jellyfin.Drawing.Skia/SkiaEncoder.cs:315]: Same for all of these

File: src/Jellyfin.Drawing.Skia/SkiaEncoder.cs (hunk 3 above)
Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/SkiaEncoder.cs:286]: this one actually not. No "Decode" call is within any other Try-Catch

File: src/Jellyfin.Drawing.Skia/SkiaEncoder.cs (hunk 4 above)
Comment (JPVenson) [reply src/Jellyfin.Drawing.Skia/SkiaEncoder.cs:315]: this one _might_ be invoked trice. But i think its not that bad.

**Review by Shadowghost — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**